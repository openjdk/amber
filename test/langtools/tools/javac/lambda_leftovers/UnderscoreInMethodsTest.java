/*
 * @test /nodynamiccopyright/
 * @bug 8173059
 * @summary Underscore for unnamed method, constructor, lambda, and catch formals
 * @compile/fail/ref=UnderscoreInMethodsTest.out -XDrawDiagnostics UnderscoreInMethodsTest.java
 */

public class UnderscoreInMethodsTest {
    class Super {
        void m(String _) {}
    }

    class Child extends Super {
        void m(String _) {
            // error '_' is not in scope so it can't be used inside the method
            System.out.println(_);
        }
    }
}
