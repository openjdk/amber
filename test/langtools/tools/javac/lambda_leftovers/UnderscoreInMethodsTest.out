UnderscoreInMethodsTest.java:10:23: compiler.err.underscore.as.identifier.in.method
UnderscoreInMethodsTest.java:14:23: compiler.err.underscore.as.identifier.in.method
UnderscoreInMethodsTest.java:16:32: compiler.err.underscore.not.allowed
3 errors
